src/GitHub.VisualStudio/source.extension.vsixmanifest

```diff
@@ -31,6 +31,8 @@
     <Asset Type="Microsoft.VisualStudio.MefComponent" d:Source="Project" d:ProjectName="GitHub.Services.Vssdk" Path="|GitHub.Services.Vssdk|" />
     <Asset Type="Microsoft.VisualStudio.MefComponent" d:Source="Project" d:ProjectName="GitHub.TeamFoundation.14" TargetVersion="[14.0,15.0)" Path="|GitHub.TeamFoundation.14|" />
     <Asset Type="Microsoft.VisualStudio.MefComponent" d:Source="Project" d:ProjectName="GitHub.TeamFoundation.15" TargetVersion="[15.0,16.0)" Path="|GitHub.TeamFoundation.15|" />
+    <!-- Sometimes servicehub's version is used (currently ~1.3) when installing for Visual Studio 2017. If the version is 1.x, assume we're installing for Visual Studio 2017.  -->
```

Comment:
Are we sure it's always servicehub's version that's being used, or is it just a random assembly that's loaded into the process? If the latter, then this won't fix the problem I don't think...

Reply:
The version is actually the version of the `VSDetouredHost` which can be found in `Common7\ServiceHub\Hosts\ServiceHub.Host.CLR.x86\` The reason this happens is that the VSDetouredHost loads a copy of the extension manager which may try to write to the extensions cache.

This fix *should* work but you'll want to test it to make sure you don't see the issue popping up still. I don't know enough about why you started using the target versions to say if this approach is better than just falling back to always attempting to load it.

Reply:
@dgriffen,

> I don't know enough about why you started using the target versions to say if this approach is better than just falling back to always attempting to load it.

The problem was that the MEF log ended up looking this [this](https://github.com/github/VisualStudio/files/927920/Microsoft.VisualStudio.Default.err.txt) when the extension installed normally. This made diagnosing MEF issues for our own and other extensions problematic (you needed to know which exceptions to ignore in order to find any genuine issues).

> The reason this happens is that the VSDetouredHost loads a copy of the extension manager which may try to write to the extensions cache.

When should I expect `ServiceHub.VSDetouredHost.exe` to run?

Reply:
> When should I expect ServiceHub.VSDetouredHost.exe to run?

The timing is tricky and I was never able to reliably produce a cache that wouldn't get detected as invalid by the next launch of VS. `VSDetouredHost` runs whenever someone requests a servicehub service that is not installed in the `Common7\ServiceHub\Services`. The most common time for this is project load. After you get it to regenerate the cache you will need to shutdown VS and delete the `ComponentModelCache` folder to trigger it to regenerate from the broken state.

Reply:
@dgriffen, thanks. I've added `How to repro` and `How to test` sections to the PR. I'm able to trigger it relatively consistently, even it this isn't exactly what is happening in the wild.